docs/NewBackendSpecificNode.md

```diff
@@ -0,0 +1,96 @@
+## New Backend-Specific Node
+
+
+This document describes how to add new backend-specific node to make further optimization on certain backend.
+
+### Graph Optimization
+
+Glow has two levels of IR. The high-level IR is a dataflow node-based graph. There are several strategies to make transformation on high-level IR to achieve better performance on certain backend. Two of these strategies will introduce new backend-specific node.
+
+#### Operator Fusion
+
+There are mainly two advantanges of operator fusion:
+
+- Good locality and less memory access
+- Reduce kernel launch time in GPU and specialized accelerators
+
+In TensorRT, the convolution, bias and ReLU layers of various sizes can be combined into a single kernel called CBR. A simple analogy is making three separate trips to the supermarket to buy three items with close relation versus buying all three in a single trip, which reduce the time of seeking items and starting trips.
+
+#### Data Layout Transformation
+
+Actually, a Tensor is a view of a block of memory. Besides a pointer to the memory, we also have to get some other descriptions of this block of memory, such as shape, stride and layout.
+
+Different layout leads to different implementation of the operator kernel on certain backend. 
+For example, in the subgraph from ResNet50, a `CPUConvDKKC8` node with memory layout modified for efficient SIMD access is introduced to optimized for CPU backend. Please refer to "5.3 Use Case: Optimizing Resnet50 for the CPU" in [Glow paper](https://arxiv.org/abs/1805.00907).
+
+We should take both fast operator kernel implementation and extra potential layout transformation into consideration to get better performance.
+
+### Steps
+
+This section is more accurately about (1) adding new backend-specific nodes and corresponding instructions, (2) how to utilize the APIs to add these backend-specifics nodes to the graph, and (3) how to have your backend correctly handle this new corresponding backend-specific instruction that is IRGen'd from your backend-specific Node.
```

Comment:
nit: you can remove "more accurately" here -- it was more of my note to you 😃

Reply:
Done